packages/common/container-definitions/src/deltas.ts

```diff
@@ -61,21 +61,94 @@ export interface IDeltaSender {
     flush(): void;
 }
 
-/** Events emitted by the Delta Manager */
+/**
+ * Events emitted by {@link IDeltaManager}.
+ */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IDeltaManagerEvents extends IEvent {
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `messageBuffer`: TODO
+     */
     (event: "prepareSend", listener: (messageBuffer: any[]) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: TODO
+     */
     (event: "submitOp", listener: (message: IDocumentMessage) => void);
+
+    /**
+     * TODO
```

Comment:
Emitted immediately after processing an incoming op.

I think it would be good to discourage customers from using this event, though I believe it has active users currently.

Reply:
What is the alternative pattern we recommend? Probably worth adding a note in the comments here.

Reply:
Depends heavily on what they're using it for - it's very low level so could be abused many ways.

It's an indicator of broken encapsulation typically - as the DeltaManager emits ops, it's expected some DDS is going to be the recipient and is the owner of the op's format, semantics, etc.  Anyone sniffing out the ops outside of that DDS is effectively spying on something that is private-ish.

So generally the advice probably sounds something like "listen to relevant events on your DDSs that are raised in response to the ops instead", but could be more complicated depending on the specifics.